processmanager/processinfo.go

```diff
@@ -525,22 +525,27 @@ func (pm *ProcessManager) processPIDExit(pid libpf.PID) {
 	pm.mu.Lock()
 	defer pm.mu.Unlock()
 
+	pidExitProcessed := false
 	info, pidExists := pm.pidToProcessInfo[pid]
-	if !pidExists {
-		log.Debugf("Skip process exit handling for unknown PID %d", pid)
-		return
-	}
-	if pm.interpreterTracerEnabled && len(pm.interpreters[pid]) > 0 {
+	if pidExists || (pm.interpreterTracerEnabled &&
+		len(pm.interpreters[pid]) > 0) {
 		// ProcessPIDExit may be called multiple times in short succession
 		// for the same PID, don't update exitKTime if we've previously recorded it.
-		if _, pidExitProcessed := pm.exitEvents[pid]; !pidExitProcessed {
+		if _, pidExitProcessed = pm.exitEvents[pid]; !pidExitProcessed {
```

Comment:
Code coverage report shows that our unit tests doesn't cover this block of code. That needs to be improved.